C/c4Document.cc

```diff
@@ -472,6 +491,90 @@ bool c4doc_save(C4Document *doc,
     return false;
 }
 
+bool c4doc_setExpiration(C4Database *db, C4Slice docId, uint64_t timestamp, C4Error *outError)
+{
+    C4Document *doc = c4doc_get(db, docId, true, outError);
+    if(doc == NULL) {
+        return false;
+    }
+    c4doc_free(doc);
+    
+    time_t now = time(NULL);
+    if(timestamp <= now) {
+        recordError(C4Domain, kC4ErrorInvalidParameter, outError);
```

Comment:
It seems like an odd thing to do though, I can see it being a mistake.  If they really want to expire a document immediately, then why not just purge it?